Documentation/Books/Manual/Programs/Arangodump/Maskings.md

```diff
@@ -99,12 +104,20 @@ Path
 ----
 
 If the path starts with a `.` then it is considered to match any path
-ending in `name`. For example, `.name` will match the attribute name
-`name` all leaf attributes in the document. Leaf attributes are
+ending in `name`. For example, `.name` will match the field
+`name` of all leaf attributes in the document. Leaf attributes are
 attributes whose value is `null` or of data type `string`, `number`,
-`bool` and `array` (see below). `name` will only match leaf attributes
+`bool` or `array`. That means, it matches `name` at the top level
+as well as at any nested level (e.g. `foo.bar.name`), but not sub-objects.
```

Comment:
I tested it and it seems to match up with the description. However, under _Nested objects and arrays_ there is the following statement:

> If the attribute value is an object, then the attribute is not masked.
Instead the **nested object is checked further for leaf attributes**.

If I have a document like:

```json
{
  "top": {
    "nested": {
      "deep": "foobar"
    },
  "other": "baz"
  }
}
```

Then a masking path of `top` does not seem to affect the string at `top.nested.deep`. So I'm not sure about the latter statement. Is it meant in the sense of "if _top_ is not a leaf attribute, then the next attribute at that level is checked", which would be _other_ in this case.

Reply:
yes, it is meant in this way. Can you suggest a better wording?

Reply:
If there's a loop over all attributes of an object, then I would consider that an implementation details which the user probably doesn't care about. How about the following?

```diff
-If the attribute value is an object, then the attribute is not masked.
-Instead the nested object is checked further for leaf attributes.
+If the attribute value is an object, then it is ignored and the attribute does not
+get masked. To mask nested fields, specify the full path for each leaf attribute.
```

What I actually miss feature-wise is a way to define a default behavior for nested attributes. For instance, what if you have a string attribute `foo` with a masking config to xify it. Then you change your data model and it becomes an object with the original value as a sub-attribute. If you forget to change the masking config, nothing will get masked. A pseudo path `"*"` could define the default masking behavior for the entire document (`"´*´"` for a path that is literally an asterisk), `"foo.*"` for the sub-attributes of `foo` (not sure about escaping, `"´foo.*"´` or `"foo.´*´"`?)

My idea might go too far. We can also add a simple warning that (nested) objects don't get masked unless explicitly configured and that users should verify that nothing sensitive is in the dump in the end.

Reply:
Let's start with the current implementation. We can later add a flag to the definition, i.e. `maskObjects: true`.